text/src/autogluon/text/text_prediction/mx/models.py

```diff
@@ -322,7 +322,7 @@ def _classification_regression_predict(net, dataloader, problem_type, label_scal
 
 
 def calculate_metric(scorer, ground_truth, predictions, problem_type):
-    if problem_type == BINARY and scorer.name == 'roc_auc':
+    if problem_type == BINARY and predictions.ndim > 1:
```

Comment:
Crashes on f1 metric. This may be a separate issue from the one being solved in this PR however, and this change merely surfaced it. @sxjscience we may in a future PR want to unify Tabular and Text's usage of metrics and calculating scores, I think they somewhat differ.

Reply:
Let me take a look. Yeah, I was thinking about Tabular vs Text but I figured we could do that after this is merged and used for a bit.

Reply:
matthews_corrcoef will also fail. I think I'll just be conservative and enable only for 'roc_auc' and 'average_precision' for now.

Reply:
I think we may first turn it on for roc_auc and average_precision. The reason is that some other metrics need the probability for pos + neg classes and do not support array that has (B,) shape.

Reply:
Great minds think alike :-). I did just that a few minutes before your comment.